### PR TITLE
Add Athena Workgroup config check

### LINKED ISSUE #150: Add new check: Athena Workgroup should enforce configuration to prevent client disabling encryption
Verify enforce Work Group Configuration is true has ‘EnforceWorkGroupConfiguration’ True

EnforceWorkGroupConfiguration: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-athena-workgroup.html
enforce_workgroup_configuration: https://www.terraform.io/docs/providers/aws/r/athena_workgroup.html

### PR DESCRIPTION
**Pull request by DrFaust92**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
Closes #150

### CODE REVIEW

File: checkov/terraform/checks/resource/aws/AthenaWorkgroupConfiguration.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        name = "Ensure Athena Workgroup should enforce configuration to prevent client disabling encryption"
+        id = "CKV_AWS_82"
+        supported_resources = ['aws_athena_workgroup']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        if 'configuration' in conf.keys():
+            if 'enforce_workgroup_configuration' in conf['configuration'][0]:
+                if conf['configuration'][0]['enforce_workgroup_configuration'] is True:
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/aws/AthenaWorkgroupConfiguration.py:16]: @DrFaust92 Can we inherit from `BaseResourceValueCheck` for this check to save the duplication of `scan_resource_conf`?

**Comment by DrFaust92** [comment]
Thanks @nimrodkor fixed and tested

**Comment by DrFaust92** [comment]
@nimrodkor, Having issues with this as the test that doesn't have the block fails (default behaviour is no block and enabled).
Should i revert? thats why i originally did this the other way.

**Comment by nimrodkor** [comment]
@DrFaust92 Great point. Just added support for overriding that case - #551 . Merge from master and it should solve your issue 😄

**Comment by DrFaust92** [comment]
Excellent, this will simplify other checks as well!

File: checkov/terraform/checks/resource/aws/AthenaWorkgroupConfiguration.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class AthenaWorkgroupConfiguration(BaseResourceValueCheck):
+
+    def __init__(self):
+        name = "Ensure Athena Workgroup should enforce configuration to prevent client disabling encryption"
+        id = "CKV_AWS_82"
+        supported_resources = ['aws_athena_workgroup']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/aws/AthenaWorkgroupConfiguration.py:12]: @DrFaust92 Missing `missing_block_result=CheckResult.PASSED`

**Comment by DrFaust92** [comment]
@nimrodkor, seems that the change you made was for the `arm` variant of the `BaseResourceValueCheck`. can you do the same for terraform?

**Review by nimrodkor — APPROVED** [review]

**Comment by DrFaust92** [comment]
Now the explicit failure test is failing that the config is valid  (when it isn't). i'll take a closer look.

**Comment by DrFaust92** [comment]
It seems to be treating `configuration/[0]/enforce_workgroup_configuration` as missing because of the `False` value and returning the default passing result.

**Comment by nimrodkor** [comment]
@DrFaust92 That was a bug in my feature, resolved in #554

**Comment by DrFaust92** [comment]
Thanks! Ready!

**Pull request merged**